src/Swaps/handlers/UniswapV2Handler.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {BaseHandler} from "./BaseHandler.sol";
+import {ISwapRouterV2} from "../vendor/ISwapRouterV2.sol";
+import {ISwapper} from "../ISwapper.sol";
+
+/// @title UniswapV2Handler
+/// @custom:security-contact security@euler.xyz
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice Swap handler executing exact output trades on Uniswap V2
+abstract contract UniswapV2Handler is BaseHandler {
+    address public immutable uniSwapRouterV2;
+
+    error UniswapV2Handler_InvalidPath();
+
+    constructor(address _uniSwapRouterV2) {
+        uniSwapRouterV2 = _uniSwapRouterV2;
+    }
+
+    /// @inheritdoc ISwapper
+    function swap(SwapParams memory params) public virtual override {
+        if (params.mode == MODE_EXACT_IN) revert Swapper_UnsupportedMode();
+        if (params.data.length < 64 || params.data.length % 32 != 0) revert UniswapV2Handler_InvalidPath();
+
+        setMaxAllowance(params.tokenIn, uniSwapRouterV2);
+        // process params according to the mode and current state
+        (uint256 amountOut, address receiver) = resolveParams(params);
+
+        if (amountOut > 0) {
+            ISwapRouterV2(uniSwapRouterV2).swapTokensForExactTokens({
+                amountOut: amountOut,
+                amountInMax: type(uint256).max,
+                path: abi.decode(params.data, (address[])),
+                to: receiver,
+                deadline: block.timestamp
```

Comment:
Deadlines should be set by the caller in `SwapParams`. As it is currently these swap txs can be held in the mempool indefinitely until they can be maximally MEV-extracted. 

Auditors will raise this as a Medium anyway.

Reply:
Ditto for other handlers

Reply:
IIRC this is/was because we are handling deadlines at a higher layer. @dglowinski - thoughts?

Reply:
Not really, I don't think. we had deadlines in original Swap, but not in the SwapHub.

What would be the impact really, when slippage is checked? Losing out on positive slippage? Do you have a specific report in mind @totomanov?

Reply:
There are dozens of findings from contest platforms and audit firms outlining this as a medium. Good analysis: https://github.com/code-423n4/2022-12-backed-findings/issues/64